### DISCUSSION TITLE
How do I validate Schematron files

### DISCUSSION THREAD

**Discussion by liu123456789** [FirelyTeam/firely-net-sdk#1777] [Q&A] [answered]
This is the official Schematron file and how I can verify it with the SDK.  Please help me!  

http://wiki.hl7.org.cn:81/static/fhir-20210610/StructureDefinition-profile-discharge-summary-document.sch

**Comment by ewoutkramer** [comment, marked as answer]
The schemas you need are included in the `specification.zip` that comes with the .NET Hl7.Fhir.Specification Nuget.

If you are *not* using the profile validator, you can directly use the .NET XML schema validation functions:

``` csharp
   XDocument docToValidate = .....;
   XmlSchemaSet xss = SchemaCollection.ValidationSchemaSet;

  ValidationEventHandler veh = (o, args) => // handle error here

  docToValidate.Validate(xss, veh);
```

If you are using the profile validator, this functionality is built-in, check this project on how to enable it: https://github.com/FirelyTeam/Furore.Fhir.ValidationDemo/blob/master/MainForm.cs#L163
